react-mutations/my-app/src/App.tsx

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useState } from 'react';
+import { Pokemon, PokemonList } from './PokemonList';
+
+export function App() {
+  const [pokedex, setPokedex] = useState([
+    { number: '001', name: 'Bulbasaur' },
+    { number: '004', name: 'Charmander' },
+    { number: '007', name: 'Squirtle' },
+    { number: '025', name: 'Pikachu' },
+    { number: '039', name: 'Jigglypuff' },
+  ]);
+
+  function handleAdd(): void {
+    setPokedex([...pokedex, { number: '012', name: 'some Pokemon' }]);
+  }
+
+  function handleUpdate(): void {
+    setPokedex(
+      pokedex.map((pokemon) =>
+        pokemon.number === '012'
+          ? { number: '002', name: 'some other Pokemon' }
```

Comment:
This is fine, but the idea behind an update is that you are modifying the _same_ item, so its ID will be the same both before and after.